Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use the .NET 8.0 SDK image to build the application
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+WORKDIR /src
+
+# Copy the solution file
+# Using a wildcard to copy the solution file, suitable if there's only one .sln file in the root
+COPY *.sln .
+
+# Copy the csproj files and restore as distinct layers
+COPY Authlete/Authlete.csproj Authlete/
+COPY Authlete.Tests/Authlete.Tests.csproj Authlete.Tests/
+
+# Restore NuGet packages
+RUN dotnet restore "Authlete.sln" -v d
+
+# Copy the rest of the source code
+COPY . .
+
+# Build the project
+RUN dotnet build "Authlete/Authlete.csproj" -c Release -o /app/build
+
+# Run tests and generate TRX test results
+RUN dotnet test "Authlete.Tests/Authlete.Tests.csproj" -c Release --no-restore --logger "trx;LogFileName=test_results.trx"
```

Comment:
Can the `dotnet test` command output the results to the console? Then there's no need to write to files and retrieve them from inside the container.